packages/message-core/src/AvMessage.js

```diff
@@ -68,10 +81,17 @@ class AvMessage {
     this.subscribers = {};
   }
 
-  onMessage(event, data) {
+  onMessage(event, data, metadata) {
+    const { isSameWindow } = metadata;
+
     if (this.subscribers[event]) {
-      for (const fn of this.subscribers[event]) {
-        fn(data);
+      for (const subscriber of this.subscribers[event]) {
+        const ignoreSameWindow = !!subscriber.options?.ignoreSameWindow;
```

Comment:
Since this is the `subscriber` object you created earlier I think it's safe to not use the `!!` or `?`.

Reply:
Good call. Fixed.